src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -698,6 +700,17 @@ class LicenceService(
     )
   }
 
+  @Transactional
+  fun removeAPConditions() {
+    val licencesInPSSPeriod = licenceRepository.getAllVariedLicencesInPSSPeriod()
+
+    licencesInPSSPeriod?.forEach { licence ->
+      val licenceWithAPConditions = licence.additionalConditions.filter { LicenceType.valueOf(it.conditionType!!) == LicenceType.AP }
```

Comment:
This will trigger an n+1 queries as it's iterating over each licence and retrieving the conditions lazily behind the scenes.

It would be good to add an integration test for this and we will be able to see this in the logs and check that improvements to the fetch strategy in the repositiory is working

Reply:
Added Integration test.